core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -301,6 +301,28 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
     if (proxyUser != null && principal != null) {
       error("Only one of --proxy-user or --principal can be provided.")
     }
+
+    // Ensure heartbeat arguments is valid.
+    val executorTimeoutMs =
+      Utils.timeStringAsMs(sparkProperties.getOrElse(
+        "spark.storage.blockManagerSlaveTimeoutMs",
+        s"${Utils.timeStringAsSeconds(
+          sparkProperties.getOrElse("spark.network.timeout", "120s"))}s"))
+    val executorHeartbeatIntervalMs = Utils.timeStringAsMs(
+      sparkProperties.getOrElse("spark.executor.heartbeatInterval", "10s"))
+    val checkTimeoutIntervalMs =
+      Utils.timeStringAsSeconds(sparkProperties.getOrElse(
+        "spark.network.timeoutInterval",
+        s"${Utils.timeStringAsMs(sparkProperties.getOrElse(
+          "spark.storage.blockManagerTimeoutIntervalMs", "60s"))}ms")) * 1000
+    if (checkTimeoutIntervalMs >= executorTimeoutMs) {
+      error(s"Incorrect heartbeat arguments, checkTimeoutIntervalMs should " +
+        s"less than executorTimeoutMs.")
+    }
+    if (executorHeartbeatIntervalMs >= checkTimeoutIntervalMs) {
```

Comment:
Overall, these configs need to use the ConfigEntry and be centralized; see https://github.com/apache/spark/pull/23447 which kind of blocks this. The problem is that this reproduces their default and parsing when it should be in one place.

@vanzin where is the best place to validate args like this, here?

Reply:
@srowen seems good! Should I will put it into the centrailized configs? [#23447](https://github.com/apache/spark/pull/23447) seems not handle heartbeat arguments.

Reply:
Yeah good point; this is what makes me wonder if this is the right place to validate this. Similar checks aren't done here in SparkSubmitArguments. For example, if these properties are mainly used in `HeartbeatReceiver`, that could be the right place to validate them.

Reply:
@srowen done! maybe I think I should put another PR to make these network arguments into the centralize config, seems some of them not appear in [#23447](https://github.com/apache/spark/pull/23447) changes.

Reply:
Yeah looks like that config is migrated in https://github.com/apache/spark/pull/23416 ; let me try to get that merged. If it's held up, we can merge this and fix it in the other PR.